src/mprisplayer.cpp

```diff
@@ -565,19 +581,24 @@ QStringList MprisPlayer::supportedMimeTypes() const
 void MprisPlayer::setServiceName(const QString &serviceName)
 {
     if (!priv->m_serviceName.isEmpty()) {
-        priv->m_connection.unregisterObject(QStringLiteral("/org/mpris/MediaPlayer2"));
-        priv->m_connection.unregisterService(priv->m_serviceName);
+        priv->m_connection->unregisterObject(QStringLiteral("/org/mpris/MediaPlayer2"));
+        priv->m_connection->unregisterService(priv->m_serviceName);
+        QDBusConnection::disconnectFromBus(priv->m_serviceName);
+        delete priv->m_connection;
+        priv->m_connection = nullptr;
     }
 
     if (!serviceName.isEmpty()) {
+        priv->m_connection = new QDBusConnection(QDBusConnection::connectToBus(dbusConnectionType(), serviceName));
```

Comment:
I'm not following. Why use separate connections? You can register as many services as you want with a single connection already?

Reply:
Because the path is always the same, QDBus doesn't have a way to know which instance is handling which service name (and in fact will refuse re-registration of the same path)

Reply:
We could try to avoid having extra connections, but I don't think it's worth the effort. D-Bus connections shouldn't be that expensive.

That would indeed be very long, as it would require a wholly new MPRIS specification.